google/cloud/storage/tests/object_integration_test.cc

```diff
@@ -679,15 +678,11 @@ TEST_F(ObjectIntegrationTest, ListObjectsFailure) {
   StatusOr<Client> client = MakeIntegrationTestClient();
   ASSERT_STATUS_OK(client);
 
-  ListObjectsReader reader = client->ListObjects(bucket_name, Versions(true));
-
   // This operation should fail because the bucket does not exist.
-  TestPermanentFailure([&] {
-    std::vector<ObjectMetadata> actual;
-    for (auto&& o : reader) {
-      actual.emplace_back(std::move(o).value());
-    }
-  });
+  ListObjectsReader reader = client->ListObjects(bucket_name, Versions(true));
+  auto it = reader.begin();
+  ASSERT_FALSE(it == reader.end());
```

Comment:
Purely opinion:  I would find `ASSERT_TRUE(it != reader.end())` easier to read.

Reply:
Done.